app/grandchallenge/components/models.py

```diff
@@ -2046,3 +2046,43 @@ def create_civ_for_file(self, *, ci, current_civ, new_value):
             # if no new value is provided (user selects '---' in dropdown)
             # delete old CIV
             self.values.remove(current_civ)
+
+
+class ValuesForInterfacesMixin:
+    @cached_property
+    def values_for_interfaces(self):
```

Comment:
This is very complex and is doing a lot of work in python. It seems like it should be possible to offload it to the database but I am unsure. I cannot really see any unit tests for this method so don't know what it needs to return.

Reply:
I can't think of a way to do this in the DB, but I'll add a test so you can see if you do.